src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -168,6 +206,42 @@ public void setDisableHttpsCertValidation(boolean disableHttpsCertValidation) {
     @CheckForNull
     public JnlpAgentEndpoint resolve() throws IOException {
         IOException firstError = null;
+
+        if (disableHttpEndpointCheck) {
```

Comment:
I find it confusing the way this new logic is inserted into this existing method, but is completely different. Especially because this is already an overly long method. At a minimum it would be nice to separate the new logic out into a separate method.

Reply:
see #337 (minimum option)